token-manager/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+#![no_std]
+
+use multiversx_sc::api::KECCAK256_RESULT_LEN;
+
+use constants::{DeployTokenManagerParams, TokenManagerType};
+use operatable::roles::Roles;
+
+use crate::constants::DEFAULT_ESDT_ISSUE_COST;
+
+multiversx_sc::imports!();
+
+pub mod constants;
+pub mod flow_limit;
+pub mod minter;
+
+#[multiversx_sc::contract]
+pub trait TokenManagerLockUnlockContract:
+    flow_limit::FlowLimit + operatable::Operatable + operatable::roles::AccountRoles + minter::Minter
+{
+    #[init]
+    fn init(
+        &self,
+        interchain_token_service: ManagedAddress,
+        implementation_type: TokenManagerType,
+        interchain_token_id: ManagedByteArray<KECCAK256_RESULT_LEN>,
+        params: DeployTokenManagerParams<Self::Api>,
+    ) {
+        require!(!interchain_token_service.is_zero(), "Zero address");
```

Comment:
I would rather not check that since the first time this is deployed it can be any address. It is much easier to do the initial deploy and testing if that check is not in place.
The initially deployed contract is only used for deploying from source in the Interchain Token Service contract, which sets it's address here anyway.